Profile Expansion and Chat Module Implementation

Shromm-Gaind opened the pull request.

jmccook96 commented on src/main/java/com/bookclub/model/ChatMessage.java:
```diff
@@ -1,49 +1,46 @@
 public class ChatMessage {
     private int messageId;
-    private int authorId;
+    private String authorUsername;
```
I was intentionally steering us away from being reliant on a username as an identifier as it makes sense for usernames to change. The suite relevant to this change gets a -1 from me.

Shromm-Gaind replied:
I did realize that authorid has Immutability, it was just easier to do, and I was on a time constraint. So I didn't bother. I can change it if you want. But it works...

jmccook96 replied:
This is part of why I encourage making all PRs specific to what it is that they tweak, it'll be less of a chore to understand where this is influenced 😆

`But it works` won't fly if it requires me to re-implement functionality because it got tweaked- sorry.

I still recommend partitioning this into seperate PRs that are relevant to their scope. Chat functionality has no relevancy to profile settings has no required relevancy to UI alteration has no required relevancy to registration. Keeping PRs neat and scoped is 90% of the game.

Potentially also suggest relinquishing chat stuff to me rather than us both beating this horse.

Shromm-Gaind replied:
Yeah I guess that's all good, will leave the chat stuff to you then. I will just focus on the profile expansion

Shromm-Gaind commented:
### User Model Updates
- Added `id`, `name`, and `email` fields to the `User` class
- Updated constructors to include new fields
- Added getters and setters for new fields

### UserDAO Updates
- Modified `findUserByUsername` to include `id`, `name`, and `email` in the query
- Updated `addUser` method to insert `name` and `email`, and return the generated `id`
- Modified `updateUser` method to update `name` and `email`, and use `id` for identification
- Added `findUserById` method to retrieve a user by their `id`
- Updated `hasUser` methods to check by both username and `id`
- Modified `deleteUser` to use `id` for identification
- Updated `tryCreateTable` to include new fields in the Users table schema

### LoginService Updates
- Modified `register` method to handle `name` and `email` during user registration
- Updated `attemptLogin` to set the current user with full details upon successful login
- Added `updateUserDetails` method to allow updating of user's name and email

### Chat Functionality
- Created `ChatMessage` model to represent individual chat messages
- Implemented `ChatDAO` for database operations related to chat messages
- Added `IChatAO` interface to define chat-related data access operations
- Created `ChatService` to manage chat-related business logic
- Implemented `ChatController` to handle UI interactions for the chat feature

### UI Updates
- Added new fields for name and email in the registration form
- Updated account settings page to display and allow editing of name and email
- Implemented a basic chat interface in JavaFX
